src/gui/qgsfeaturelistcombobox.cpp

```diff
@@ -136,6 +137,18 @@ void QgsFeatureListComboBox::restoreLineEditState()
     mLineEditState.restore( mLineEdit );
 }
 
+int QgsFeatureListComboBox::nullIndex() const
+{
+  int index = -1;
+
+  if ( allowNull() )
+  {
+    index = findText( tr( "NULL" ) );
```

Comment:
Is this reliable if there's an item with the text `NULL` but which is *not* the NULL value?

Reply:
Considering that `findText` will search for the first element matching NULL text and that the "true" NULL value is always added ahead of others items, I think so.